doc/examples/boilerplate.py

```diff
@@ -0,0 +1,135 @@
+"""
+A docstring
+
+This can double as a helpful message which explains how the program is run.
+"""
+
+from __future__ import absolute_import, division, print_function
+
+import logging
+import sys
+from typing import List  # noqa: F401  Flake8 doesn't recognise Python 2 typing.
+
+import libtbx.phil
+from dials.array_family import flex
+import dials.util
+import dials.util.log
+from dials.util.options import OptionParser, flatten_reflections, flatten_experiments
+from dxtbx.model import ExperimentList
+
+
+# Define a logger
+logger = logging.getLogger("dials.logger_name")
+
+# Define the master PHIL scope for this program
+phil_scope = libtbx.phil.parse(
+    """
+    output {
+        reflections = stronger.refl
+            .type = path
+            .help = "Help strings are helpful, wherever possible."
+        log = dials.command_name.log
+            .type = path
+    }
+    bool_parameter = False
+      .type = bool
+    integer_parameter = 0
+      .type = int
+    """
+)
+
+
+def do_stuff(experiments, reflections, params):
```

Comment:
Propose we state clearly here that `do_stuff` is _renamed_ otherwise we will be defining `do_stuff` methods everywhere and it will be about as useful as `run()`

Reply:
I suggest that we keep consistent with this - `boilerplate.py`, so `do_boilerplate`, `getLogger("dials.boilerplate")` (assuming no better way to get dispatcher entry point name?) etc. Basically, so that someone can search-and-replace to get a template rather than having to scan over everything for placeholdery words

Reply:
Yes, good point, I think you're right.  I'll make a change.